src/relations/osm_mysql.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library containing the implementation of the legacy osm-mysql relation."""
+
+import json
+import logging
+
+from ops.charm import CharmBase, RelationBrokenEvent, RelationCreatedEvent
+from ops.framework import Object
+from ops.model import BlockedStatus
+
+from constants import LEGACY_OSM_MYSQL, PASSWORD_LENGTH
+from utils import generate_random_password
+
+logger = logging.getLogger(__name__)
+
+
+class MySQLOSMRelation(Object):
+    """Encapsulation of the legacy osm-mysql relation."""
+
+    def __init__(self, charm: CharmBase):
+        super().__init__(charm, LEGACY_OSM_MYSQL)
+
+        self.charm = charm
+
+        self.framework.observe(self.charm.on.leader_elected, self._on_leader_elected)
+        self.framework.observe(
+            self.charm.on[LEGACY_OSM_MYSQL].relation_created, self._on_osm_mysql_relation_created
+        )
+        self.framework.observe(
+            self.charm.on[LEGACY_OSM_MYSQL].relation_broken, self._on_osm_mysql_relation_broken
+        )
+
+    def _get_or_set_password_in_peer_databag(self, username: str) -> str:
```

Comment:
`_get_password_in_peer_databag `

Reply:
addressed in 2f6e6e434f2bab7f2a508153eb234fb73edf2bed